client/db/bolt/upgrades.go

```diff
@@ -247,6 +250,70 @@ func reloadMatchProofs(tx *bbolt.Tx, skipCancels bool) error {
 	})
 }
 
+// v4Upgrade moves active orders from what will become the archivedOrdersBucket
+// to a new ordersBucket. This is done in order to make searching active orders
+// faster, as they do not need to be pulled out of all orders any longer. This
+// upgrade moves active orders as apposed to inactive orders under the
+// assumption that there are less active orders to move, and so a smaller
+// database transaction occurs.
+func v4Upgrade(dbtx *bbolt.Tx) error {
+	const oldVersion = 3
+
+	if err := ensureVersion(dbtx, oldVersion); err != nil {
+		return err
+	}
+
+	// Move any executed orders to the new archivedOrdersBucket.
+	return moveActiveOrders(dbtx)
+}
+
+// moveActiveOrders searches the v1 ordersBucket for orders that are not yet
+// booked booked status, adds those to the v2 ordersBucket, and deletes them
+// from the v1 ordersBucket.
+func moveActiveOrders(tx *bbolt.Tx) error {
+	ordersBucketV1 := []byte("orders")
+	ordersBucketV2 := []byte("activeOrders")
+	// NOTE: orderBucketV2 created in NewDB, but TestUpgrades skips that.
+	_, err := tx.CreateBucketIfNotExists(ordersBucketV2)
+	if err != nil {
+		return err
+	}
+
+	ordersBktV1 := tx.Bucket(ordersBucketV1)
+	ordersBktV2 := tx.Bucket(ordersBucketV2)
+	var (
+		oBktV1, oBktV2 *bbolt.Bucket
+		status         order.OrderStatus
+	)
```

Comment:
How come declared outside the `ForEach` closure?  I tend to prefer limiting scope.

Reply:
Limited.